thunder/clang/__init__.py

```diff
@@ -2055,6 +2055,28 @@ def zeta(a, b):
     )
 
 
+#
+# Elementwise ternary operations
+#
+
+
+@clangop()
+def lerp(start, end, weight):
+    inputs = (start, end, weight)
+    # torch.lerp does not promote types and only accepts floating-point inputs
```

Comment:
Does this suggest that the type promotion kind should be `INT_TO_FLOAT`?

https://github.com/Lightning-AI/lightning-thunder/blob/ceba7307d1ec6c43683a5f5bcf376dfb1841d240/thunder/core/utils.py#L411

Reply:
The original `torch.lerp` raises a `RuntimeError` when the inputs are integers. I tried to follow that behavior, but it is possible to choose `INT_TO_FLOAT` and support more input patterns.